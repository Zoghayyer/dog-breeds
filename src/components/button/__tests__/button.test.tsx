import React from 'react';
import { shallow } from 'enzyme';
import Button, { ButtonProps } from '../button';

describe('(Component) Button', () => {
  let component;

  const defaultProps: ButtonProps = {
    children: <div>child</div>,
    className: 'className',
    onClick: () => ({}),
    size: 'sm',
    theme: 'primary',
    type: 'button',
  };

  const shallowButton = (extendedProps: Partial<ButtonProps> = {}) => {
    component = shallow(<Button {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowButton());

  it('renders without crashing', () => {
    shallow(<Button {...defaultProps} />);
  });

  it('is rendered with primary class attributes when passed `primary` as a theme', () => {
    shallowButton({theme: 'primary'});

    expect(component).toMatchSnapshot();
  });

  it('is rendered with secondary class attributes when passed `secondary` as a theme', () => {
    shallowButton({ theme: 'secondary' });

    expect(component).toMatchSnapshot();
  });

  it('is rendered with tertiary class attributes when passed `tertiary` as a theme', () => {
    shallowButton({ theme: 'tertiary' });

    expect(component).toMatchSnapshot();
  });

  it('renders a large button when passed size = `lg`', () => {
    shallowButton({ size: 'lg' });

    expect(component).toMatchSnapshot();
  });

  it('calls onClick on button click', () => {
    const onClickSpy = jest.fn();
    shallowButton({ onClick: onClickSpy });

    component.find('button').simulate('click');
    expect(onClickSpy).toHaveBeenCalledTimes(1);
  });
});
