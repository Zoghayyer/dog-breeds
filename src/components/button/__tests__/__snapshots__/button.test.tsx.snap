// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Button is rendered with primary class attributes when passed \`primary\` as a theme 1`] = `
<button
  className="btn btn-primary btn-sm className"
  onClick={[Function]}
  type="button"
>
  <div>
    child
  </div>
</button>
`;

exports[`(Component) Button is rendered with secondary class attributes when passed \`secondary\` as a theme 1`] = `
<button
  className="btn btn-secondary btn-sm className"
  onClick={[Function]}
  type="button"
>
  <div>
    child
  </div>
</button>
`;

exports[`(Component) Button is rendered with tertiary class attributes when passed \`tertiary\` as a theme 1`] = `
<button
  className="btn btn-tertiary btn-sm className"
  onClick={[Function]}
  type="button"
>
  <div>
    child
  </div>
</button>
`;

exports[`(Component) Button renders a large button when passed size = \`lg\` 1`] = `
<button
  className="btn btn-primary btn-lg className"
  onClick={[Function]}
  type="button"
>
  <div>
    child
  </div>
</button>
`;
